src/test/java/reactor/core/publisher/FluxPublishTest.java

```diff
@@ -112,7 +112,7 @@ public void cancelComposes2() {
 	public void pairWise() {
 		TestSubscriber<Tuple2<Integer, Integer>> ts = TestSubscriber.create();
 
-		range(1, 9).compose(o -> zip(o, o.skip(1)))
+		range(1, 9).transform(o -> zip(o, o.skip(1)))
```

Comment:
I'd keep the original compose version as well and have a separate test for this. In addition, I'd test if the function is executed the time the transform is applied an not when the subscription happens.